bumble/device.py

```diff
@@ -5789,6 +5792,18 @@ async def reply() -> None:
 
         utils.AsyncRunner.spawn(reply())
 
+    # [Classic only]
+    @host_event_handler
+    @with_connection_from_address
+    def on_mode_change(self, connection, status, current_mode, interval):
+        # if status is success: set current_mode and interval
+        # else: emit mode change fail
+        if status != hci.HCI_SUCCESS:
+            self.classic_mode = current_mode
```

Comment:
`classic_mode` and `classic_interval` are properties of the connection, not of the `Device` object.